### PR TITLE
Use next-runtime-env for dynamic envs

### PR DESCRIPTION
**Pull request by MikeShi42**
Disabling ASO had some issues with app navigation clashing with query parameter updates from all our nuqs usage. Switching to `next-runtime-env` and also changing the preset environment variables to `NEXT_PUBLIC_HDX_LOCAL_DEFAULT_CONNECTIONS` and `NEXT_PUBLIC_HDX_LOCAL_DEFAULT_SOURCES`, which is required for the package, but also makes it nice and clear that these are going to be publicly exposed environment variables.

### CODE REVIEW

**Comment by jokester** [comment]
Hi I'm just a random guy investigated a few runtime configuration issues today.

For curiosity I want to ask, can it be fixed with [next.js runtime config](https://nextjs.org/docs/pages/api-reference/next-config-js/runtime-configuration) without introducing a new dependency?

**Comment by MikeShi42** [comment]
@jokester I suspect it'll be an app-specific answer. One of our users opened up https://github.com/hyperdxio/hyperdx/pull/498 initially so we did try that approach, he documents well the drawbacks of using the runtime config (need to disable ASO, can't use standalone builds, can't use RSC... etc). While we did use ASO (it's the default behavior in Next) and we had a build that used standalone builds - we thought we could workaround those issues.

In the end disabling ASO (by creating dummy server side props) caused issues with our `<Link>`-based components navigating throughout the app. I suspect disabling ASO caused navigations to get delayed, and we use `nuqs` liberally across the app which programmatically interfaces with the Next router. There were a lot of calls across the app from `nuqs` that would race the `Link` navigation and cause the user's navigation to cancel in favor of some other random programmatic route update from `nuqs`. (ex. Live Tail updating query parameters)

I couldn't see a cleaner way to disable ASO while avoiding the weird side effects it caused, so the two approaches we had left were to either go down a similar route to `next-runtime-env` (include a script file that's generated at run time) or a route similar to https://github.com/hyperdxio/hyperdx/issues/462#issuecomment-2499140445 where we find/replace the string at the Docker entrypoint. The former was easier to ship at the moment so we chose that path.

Hope that helps!

**Comment by jokester** [comment]
So use of `next-runtime-env` is to enable ASO and runtime configuration at the same time? I think I got it. Thanks!!

**Comment by jokester** [comment]
One related thing that confused me was the different behavior of `/api/config` across 2 docker images [detail](https://github.com/hyperdxio/hyperdx/issues/462#issuecomment-2508913744) .

But I guess with `next-runtime-env` the API can even be retired so no problem 👍🏽

File: packages/app/src/config.ts [hunk 1]
```diff
@@ -11,10 +9,12 @@ export const CLICKHOUSE_HOST =
...
...},"timestampValueExpression":"Timestamp","defaultTableSelectExpression":"Timestamp, ServiceName, StatusCode, round(Duration / 1e6), SpanName","serviceNameExpression":"ServiceName","eventAttributesExpression":"SpanAttributes","resourceAttributesExpression":"ResourceAttributes","traceIdExpression":"TraceId","spanIdExpression":"SpanId","implicitColumnExpression":"SpanName","durationExpression":"Duration","durationPrecision":9,"parentSpanIdExpression":"ParentSpanId","spanKindExpression":"SpanKind","spanNameExpression":"SpanName","logSourceId":"l-758211293","statusCodeExpression":"StatusCode","statusMessageExpression":"StatusMessage"},{"id":"l-758211293","kind":"log","name":"Demo Logs","connection":"local","from":{"databaseName":"default","tableName":"otel_logs"},"timestampValueExpression":"TimestampTime","defaultTableSelectExpression":"Timestamp, ServiceName, SeverityText, Body","serviceNameExpression":"ServiceName","severityTextExpression":"SeverityText","eventAttributesExpression":"LogAttributes","resourceAttributesExpression":"ResourceAttributes","traceIdExpression":"TraceId","spanIdExpression":"SpanId","implicitColumnExpression":"Body","traceSourceId":"l701179602"}]' yarn dev:local
```
Comment (wrn14897) [inline packages/app/src/config.ts:11]: nit: would be great to update this comment as well

File: packages/app/src/config.ts [hunk 2]
```diff
@@ -11,10 +9,12 @@ export const CLICKHOUSE_HOST =
...
-export const HDX_LOCAL_DEFAULT_CONNECTIONS =
-  publicRuntimeConfig.hdxLocalDefaultConnections;
-export const HDX_LOCAL_DEFAULT_SOURCES =
-  publicRuntimeConfig.hdxLocalDefaultSources;
+export const HDX_LOCAL_DEFAULT_CONNECTIONS = env(
+  'NEXT_PUBLIC_HDX_LOCAL_DEFAULT_CONNECTIONS',
+);
+export const HDX_LOCAL_DEFAULT_SOURCES = env(
```
Comment (wrn14897) [inline packages/app/src/config.ts:15]: I feel it would be handy if the `NEXT_PUBLIC_IS_LOCAL_MODE` is also runtime

**Review by wrn14897 — DISMISSED** [review]
Thanks for the fix! Tested it on my local and it works like a charm

**Comment by wrn14897** [comment]
nit: we probably want to add `public/__ENV.js` to gitignore

File: packages/app/src/config.ts (hunk 2 above)
Comment (MikeShi42) [reply packages/app/src/config.ts:15]: I'll leave that to a separate PR, since I think we do different images/serving modes for that anyways.

**Comment by MikeShi42** [comment]
@wrn14897 good point, updated the gitignore

**Review by wrn14897 — APPROVED** [review]

**Pull request merged**